include/capstone/loongarch.h

```diff
@@ -0,0 +1,2516 @@
+#ifndef CAPSTONE_LOONGARCH_H
+#define CAPSTONE_LOONGARCH_H
+
+/* Capstone Disassembly Engine */
+/* By Jiajie Chen <c@jia.je> 2024 */
+/*    Yanglin Xun <1109673069@qq.com>, 2024 */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !defined(_MSC_VER) || !defined(_KERNEL_MODE)
+#include <stdint.h>
+#endif
+
+#include "platform.h"
+#include "cs_operand.h"
+
+/// Operand type for instruction's operands
+typedef enum loongarch_op_type {
+	LoongArch_OP_INVALID = CS_OP_INVALID, ///< Invalid
+	LoongArch_OP_REG = CS_OP_REG,	      ///< Register operand
+	LoongArch_OP_IMM = CS_OP_IMM,	      ///< Immediate operand
+	LoongArch_OP_MEM = CS_OP_MEM,	      ///< Memory operand
+} loongarch_op_type;
+
+/// Instruction's operand referring to memory
+/// This is associated with LOONGARCH_OP_MEM operand type above
+typedef struct loongarch_op_mem {
+	unsigned int base;  ///< base register
+	unsigned int index; ///< index register
+	int64_t disp;	    ///< displacement/offset value
+} loongarch_op_mem;
+
+/// Instruction operand
+typedef struct cs_loongarch_op {
+	loongarch_op_type type;	      //< operand type
+	union {
+		unsigned int reg;     /// register value for REG operand
+		int64_t imm;	      /// immediate value for IMM operand
+		loongarch_op_mem mem; /// base/disp value for MEM operand
+	};
+
+	/// How is this operand accessed? (READ, WRITE or READ|WRITE)
+	/// NOTE: this field is irrelevant if engine is compiled in DIET mode.
+	uint8_t access;
+} cs_loongarch_op;
+
+/// LoongArch instruction formats. To get details about them please
+/// refer to `LoongArchInstrFormats.td` in LLVM.
+typedef enum {
+	// generated content <LoongArchGenCSInsnFormatsEnum.inc> begin
+	// clang-format off
+
+LoongArch_INSN_FORM_PSEUDO,
```

Comment:
please indent this line

Reply:
and we need capitalized letters too.

Reply:
> and we need capitalized letters too.

The name is concatenated like `${ARCH}_INSN_FORM_${FORM}`, where `ARCH` comes from LLVM which uses `LoongArch` instead of all-capitalized letters of `LOONGARCH`. @Rot127 What's your opinion on this? We will need to change the capstone printer in llvm if required.

Reply:
Just checked it with the other new archs. TriCore is all capitals as well (but not generated with Auto-Sync). I can remember that we had this discussion, but can't find it.
Good chance I wanted to change it later and forgot it?
@jiegec I am currently off and won't be able to take care of it in the next week. If you find time, feel free to fix up `PrinterCapstone`. Or, if it isn't easily done, let us know. It is possible that I didn't generate CS enums in capital letters for a reason (can't remember, though).
If you are busy, no worries. I'll update it in roughly a week.

Reply:
Thanks @Rot127, I have merged your changes